pkg/summarizer/flakiness.go

```diff
@@ -0,0 +1,286 @@
+package summarizer
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/testgrid/internal/result"
+	"github.com/GoogleCloudPlatform/testgrid/pb/state"
+	"github.com/sirupsen/logrus"
+	"vbom.ml/util/sortorder"
+)
+
+var INFRA_REGEX = regexp.MustCompile(`^\w+$`)
+var JAILED_REGEX = regexp.MustCompile(`(JAILED )?(\/?\/?[A-Za-z0-9\/_\-.:}{]+) - \[([A-Za-z0-9\/:\-_.\s]+)\]\s*`)
+
+// Keep in mind that flakiness is measured as out of 100, i.e. 23 not .23
+var DEFAULT_FLAKINESS = 50.0
+var MIN_RUNS = 0
+var HEALTHY_RANGE = map[int]string{
+	20: "red",
+	10: "purple",
+	3:  "orange",
+	0:  "green",
+}
+
+type Result struct {
+	name             string
+	passed           int
+	failed           int
+	flakyCount       int
+	averageFlakiness float64
+	failedInfraCount int
+	infraFailures    map[string]int
+}
+
+// Temporary structs while I decide what will go into summary.proto
+type Healthiness struct {
+	startDate        int
```

Comment:
I have seen these mostly as floats in other places and maybe `Timestamp` over `Date`

Reply:
I believe Timestamp is the canonical way for new fields. (We use doubles or floats some places, but they should be timestamps).

Reply:
Done